include/openthread/platform/crypto.h

```diff
@@ -469,6 +469,36 @@ otError otPlatCryptoSha256Update(otCryptoContext *aContext, const void *aBuf, ui
  */
 otError otPlatCryptoSha256Finish(otCryptoContext *aContext, uint8_t *aHash, uint16_t aHashSize);
 
+/**
+ * Initialize cryptographically-secure pseudorandom number generator (CSPRNG).
+ *
+ * @retval OT_ERROR_NONE     Successfully initialized CSPRNG module.
+ * @retval OT_ERROR_FAILED   Failed to initialize CSPRNG module.
+ *
+ */
+otError otPlatCryptoRandomInit(void);
```

Comment:
As currently for non-TF-M enabled platforms there is only one global `ctr_drbg` and `entropy` context, I decided to not introduce `otCryptoContext` parameter to the API, and keep it in `crypto_platfrom.cpp` (for default configuration). However, if you see the use case to pass a context to the platform, I'm fine to introduce this change.

Reply:
The only advantage I see with adding `otCryptoContext` is that we can get rid of the contexts and the default mbedtls implementation we have in `crypto_platform.cpp` We do already have a platform abstraction for entropy, with this we can hand over the `otPlatCryptoRandomInit` `otPlatCryptoRandomDeinit` and `otPlatCryptoRandomGet` to vendors. They can maintain the contexts and the callbacks as needed. 

We can add the default implementation to entropy.cpp instead for posix builds.

Reply:
Discussed on slack to change the ifdef condition in `crypto_platform.cpp`. Will be done in the next commit. Thanks!